File: qqbot-domain/src/main/java/com/mieai/qqbot/domain/bot/BotEnvironment.java
package com.mieai.qqbot.domain.bot;

import java.util.Locale;
import java.util.Objects;

/** QQ environment whose sessions and target identifiers must remain isolated. */
public enum BotEnvironment {
    SANDBOX,
    PRODUCTION;

    public static BotEnvironment parse(String value) {
        Objects.requireNonNull(value, "value must not be null");
        if (value.isBlank()) {
            throw new IllegalArgumentException("value must not be blank");
        }

        try {
            return valueOf(value.strip().toUpperCase(Locale.ROOT));
        } catch (IllegalArgumentException exception) {
            throw new IllegalArgumentException("unsupported bot environment: " + value, exception);
        }
    }

    public boolean isSandbox() {
        return this == SANDBOX;
    }
}

File: qqbot-domain/src/main/kotlin/com/mieai/qqbot/domain/bot/BotEnvironment.kt
package com.mieai.qqbot.domain.bot

import java.util.Locale

/** QQ environment whose sessions and target identifiers must remain isolated. */
enum class BotEnvironment {
    SANDBOX,
    PRODUCTION,
    ;

    fun isSandbox(): Boolean = this == SANDBOX

    companion object {
        fun parse(value: String): BotEnvironment {
            require(value.isNotBlank()) { "value must not be blank" }

            return try {
                valueOf(value.trim().uppercase(Locale.ROOT))
            } catch (exception: IllegalArgumentException) {
                throw IllegalArgumentException("unsupported bot environment: $value", exception)
            }
        }
    }
}
